Feature/nav bar

nk-radikant opened the pull request:
The main changes done in this PR are:

what: added header tag for nav-bars
why: to have more semantically correct nav-bars

what: fixed the ocean borders position for desktop (Extended Level 2, Extended Level 3) and mobile (Default, Extended, Extended Level 2, Extended Level 3)
why: so that it looks like the designers intended in Figma

what: divided desktop nav-bar menus into smaller reusable template parts and changed those template names
why: for a better overview and maintainability

Other than that I fixed font-weight for dropdown hover behaviour, replaced some fixed values in scss with our tokens, and deleted the data-text attribute in the nav-bar links because it got redundant.
<!-- GITHUB_RELEASE PR BODY: canary-version -->
<details>
  <summary>📦 Published PR as canary version: <code>1.3.0--canary.26.f5eb94067647508cd35c5b96e2e19aaa70d87014.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @infineon/design-system-bootstrap@1.3.0--canary.26.f5eb94067647508cd35c5b96e2e19aaa70d87014.0
  # or 
  yarn add @infineon/design-system-bootstrap@1.3.0--canary.26.f5eb94067647508cd35c5b96e2e19aaa70d87014.0
  ```
</details>
<!-- GITHUB_RELEASE PR BODY: canary-version -->

nk-radikant committed:
fix mobile border + add <header>

nk-radikant committed:
fix dropdown hover font-weight

nk-radikant committed:
fix desktop border

nk-radikant committed:
scss fixed values ​​replaced with tokens

nk-radikant committed:
add templates for desktop Nav-Bar

nk-radikant committed:
delete data-text attribute

nk-radikant committed:
add more templates for desktop Nav-Bar
+ change template names

tishoyanchev reviewed (changes requested):
@nk-radikant Please resolve merge conflicts, and then add a short description of the changes you have made.

kaiwerther committed:
Merge remote-tracking branch 'origin/master' into feature/nav-bar

# Conflicts:
#	src/nav-bar/_nav-bar.scss

kaiwerther commented:
I merged master into this release to get GitHub Actions working - will forward the resulting URL to the designers for feedback

kaiwerther commented:
Please label PRs directly as release + one of major/minor/patch

kaiwerther commented:
DDS-13 + DDS-14 seems to be fixed, I close them.

@nk-radikant Can you please describe what the issue was and how you fixed it? It isn't clear at all from the source code changes.

nk-radikant commented:
The main changes done in this PR are:

**what**: added header tag for nav-bars
**why**: to have more semantically correct nav-bars 

**what**: fixed the ocean borders position for desktop (Extended Level 2, Extended Level 3)  and mobile (Default, Extended, Extended Level 2, Extended Level 3)
**why**: so that it looks like the designers intended in Figma

**what**: divided desktop nav-bar menus into smaller reusable template parts and changed those template names
**why**: for a better overview and maintainability

Other than that I fixed font-weight for dropdown hover behaviour, replaced some fixed values in scss with our tokens, and deleted the data-text attribute in the nav-bar links because it got redundant.

benedikt-ifx commented:
Visual changes are fine and approved.

tishoyanchev reviewed (changes requested):
Navbar overflows on Extended versions.
Please remove the 110px from margin-left on .ifx__nav-link-wrapper.ifx__desktop-show class.
Double check before submitting another PR.

kaiwerther commented:
![image](https://user-images.githubusercontent.com/21239573/192274282-8c20d23c-c7ba-480f-8219-18eded96087b.png)

nk-radikant committed:
fix overflow for extended nav-bars
